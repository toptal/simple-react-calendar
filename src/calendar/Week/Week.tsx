import React, { FC, SyntheticEvent } from 'react'

import { IDateRange, IWeekDayRenderProps, IWeekRenderProps } from '../../@types'
import WeekDay from '../WeekDay'

export type Props = {
  blockClassName: string
  customRender?: IWeekRenderProps
  getDayEachDay: ({start, end}: IDateRange) => Date[]
  getDayFormatted: (date: Date) => string
  getDayIsCurrentMonth: (date: Date) => boolean
  getDayIsDisabled: (date: Date) => boolean
  getDayIsHighlighted: (date: Date) => boolean
  getDayIsNextMonth: (date: Date) => boolean
  getDayISOFormatter: (date: Date) => string
  getDayIsSelectable: (date: Date) => boolean
  getDayIsSelected: (date: Date) => boolean
  getDayIsSelectedEnd: (date: Date) => boolean
  getDayIsSelectedStart: (date: Date) => boolean
  getDayIsToday: (date: Date) => boolean
  getDayIsWeekend: (date: Date) => boolean
  handleOnDayClick: (event: SyntheticEvent<HTMLButtonElement>) => void
  handleOnDayMouseEnter: (event: SyntheticEvent<HTMLButtonElement>) => void
  handleOnDisabledDayClick: (event: SyntheticEvent<HTMLButtonElement>) => void
  renderWeekDay?: IWeekDayRenderProps
  weekEnd: Date
  weekNumber: number
  weekStart: Date
}

const Week: FC<Props> = (props) => {
  const {
    blockClassName,
    customRender,
    getDayEachDay,
    getDayFormatted,
    getDayIsCurrentMonth,
    getDayIsDisabled,
    getDayIsHighlighted,
    getDayIsNextMonth,
    getDayISOFormatter,
    getDayIsSelectable,
    getDayIsSelected,
    getDayIsSelectedEnd,
    getDayIsSelectedStart,
    getDayIsToday,
    getDayIsWeekend,
    handleOnDayClick,
    handleOnDayMouseEnter,
    handleOnDisabledDayClick,
    renderWeekDay,
    weekEnd,
    weekStart,
  } = props

  const children = getDayEachDay({start: weekStart, end: weekEnd}).map((day: Date) => {
    const date = getDayISOFormatter(day)
    const isCurrentMonth = getDayIsCurrentMonth(day)
    const isDisabled = getDayIsDisabled(day)
    const isNextMonth = !isCurrentMonth && getDayIsNextMonth(day)
    const isSelectable = getDayIsSelectable(day)
    const isWorkDay = !getDayIsWeekend(day)

    return (
      <WeekDay
        blockClassName={blockClassName}
        customRender={renderWeekDay}
        date={date}
        getDayFormatted={getDayFormatted}
        handleOnClick={isSelectable ? handleOnDayClick : isDisabled ? handleOnDisabledDayClick : undefined}
        handleOnEnter={isSelectable ? handleOnDayMouseEnter : undefined}
        isCurrentMonth={getDayIsCurrentMonth(day)}
        isDisabled={isDisabled}
        isHighlighted={getDayIsHighlighted(day)}
        isMonthNext={isNextMonth}
        isMonthPrev={!isCurrentMonth && !isNextMonth}
        isNonSelectable={!isSelectable}
        isSelectable={isSelectable}
        isSelected={getDayIsSelected(day)}
        isSelectionEnd={getDayIsSelectedEnd(day)}
        isSelectionStart={getDayIsSelectedStart(day)}
        isToday={getDayIsToday(day)}
        isWeekend={!isWorkDay}
        isWorkday={isWorkDay}
        key={date}
      />
    )
  })

  if (customRender) {
    return customRender(props)
  }

  return <div className={`${blockClassName}-week`}>{children}</div>
}

Week.displayName = 'Week'

export default Week
