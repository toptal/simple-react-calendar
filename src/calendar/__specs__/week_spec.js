import React from 'react'

import ReactTestRenderer from 'react-test-renderer'
import Week from '../week'

import {shallow} from 'enzyme'

describe.only('Week', () => {
  let props, tree
  const defaultProps = {
    date: new Date(2015, 7, 17),
    today: new Date(2015, 7, 17),
    activeMonth: new Date(2015, 7, 17),
    weekStartsOn: 1,
    onDayClick: () => {},
    onDayMouseMove: () => {},
    onDisabledDayClick: () => {},
  }

  describe('#render', () => {
    it('renders <Week />', () => {
      tree = ReactTestRenderer.create(<Week {...defaultProps} />).toJSON()

      expect(tree).toMatchSnapshot()
    })

    context('when prop `date` is different month', () => {
      it('renders <Week />', () => {
        tree = ReactTestRenderer.create(
          <Week {...Object.assign(defaultProps, {date: new Date(2015, 5, 29)})} />
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    context('when prop `date` falls in prop `activeMonth`', () => {
      it('renders <Week />', () => {
        props = Object.assign(defaultProps, {
          date: new Date(2015, 5, 29),
          activeMonth: new Date(2015, 5, 1),
        })
        tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    context('when prop `date` falls in the previous month', () => {
      it('renders <Week />', () => {
        props = Object.assign(defaultProps, {
          date: new Date(2015, 6, 27),
          activeMonth: new Date(2015, 7, 1),
        })
        tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    context('when prop `highlightedStart` and `highlightedEnd` are defined', () => {
      it('renders <Week />', () => {
        props = Object.assign(defaultProps, {
          date: new Date(2015, 5, 29),
          today: new Date(2015, 5, 31),
          activeMonth: new Date(2015, 5, 1),
          highlightedStart: new Date(2015, 5, 29),
          highlightedEnd: new Date(2015, 6, 1),
        })
        tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    context('when prop `selectedMin` and `selectedMax` are defined', () => {
      it('renders <Week />', () => {
        props = Object.assign(defaultProps, {
          date: new Date(2015, 5, 29),
          activeMonth: new Date(2015, 5, 1),
          selectedMin: new Date(2015, 5, 30),
          selectedMax: new Date(2015, 6, 2),
        })
        tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
      })

      context('when prop `selectedMin` and `selectedMax` are the same day', () => {
        it('renders <Week />', () => {
          props = Object.assign(defaultProps, {
            date: new Date(2015, 5, 29, 21, 0),
            activeMonth: new Date(2015, 5, 1),
            selectedMin: new Date(2015, 5, 30),
            selectedMax: new Date(2015, 5, 30),
          })
          tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

          expect(tree).toMatchSnapshot()
        })
      })
    })

    context('when prop `minDate` is defined', () => {
      it('renders <Week />', () => {
        props = Object.assign(defaultProps, {
          date: new Date(2015, 5, 29),
          activeMonth: new Date(2015, 5, 1),
          minDate: new Date(2015, 5, 30),
        })
        tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    context('when prop `maxDate` is defined', () => {
      it('renders <Week />', () => {
        props = Object.assign(defaultProps, {
          date: new Date(2015, 5, 29),
          activeMonth: new Date(2015, 5, 1),
          maxDate: new Date(2015, 5, 30),
        })
        tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    context('when prop `selectedMin` and `selectedMax` are defined', () => {
      it('renders <Week />', () => {
        props = Object.assign(defaultProps, {
          date: new Date(2015, 5, 29),
          activeMonth: new Date(2015, 5, 1),
          minDate: new Date(2015, 5, 30),
          maxDate: new Date(2015, 6, 1),
        })
        tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    context('when prop `blockClassName` is defined', () => {
      it('renders el with prefixed class name', () => {
        props = Object.assign(defaultProps, {blockClassName: 'cal'})
        tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    context('when prop `disabledIntervals` is defined', () => {
      const disabledIntervals = [
        {
          start: new Date(2015, 7, 21),
          end: new Date(2015, 7, 23),
        },
      ]
      props = Object.assign(defaultProps, {disabledIntervals})

      it('renders <Week />', () => {
        tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    context('when prop `dateDisabled`', () => {
      it('renders <Week />', () => {
        props = Object.assign(defaultProps, {
          date: new Date(2015, 5, 29),
          activeMonth: new Date(2015, 5, 1),
          minDate: new Date(2015, 5, 30),
          maxDate: new Date(2015, 6, 1),
        })
        tree = ReactTestRenderer.create(<Week {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })
  })
})
