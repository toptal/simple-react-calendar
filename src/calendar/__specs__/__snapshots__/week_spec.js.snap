// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Week #render renders <Week /> 1`] = `
<div
  className="calendar-week"
>
  <div
    className="calendar-day is-today is-current_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    17
  </div>
  <div
    className="calendar-day is-current_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    18
  </div>
  <div
    className="calendar-day is-current_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    19
  </div>
  <div
    className="calendar-day is-current_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    20
  </div>
  <div
    className="calendar-day is-disabled is-current_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    21
  </div>
  <div
    className="calendar-day is-disabled is-current_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    22
  </div>
  <div
    className="calendar-day is-disabled is-current_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    23
  </div>
</div>
`

exports[`Week #render when prop \`blockClassName\` is defined renders el with prefixed class name 1`] = `
<div
  className="cal-week"
>
  <div
    className="cal-day is-highlighted is-current_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="cal-day is-selected is-highlighted is-current_month is-start_selection is-end_selection is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="cal-day is-highlighted is-today is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="cal-day is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="cal-day is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="cal-day is-next_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="cal-day is-next_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`

exports[`Week #render when prop \`date\` falls in prop \`activeMonth\` renders <Week /> 1`] = `
<div
  className="calendar-week"
>
  <div
    className="calendar-day is-current_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="calendar-day is-current_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`

exports[`Week #render when prop \`date\` falls in the previous month renders <Week /> 1`] = `
<div
  className="calendar-week"
>
  <div
    className="calendar-day is-prev_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    27
  </div>
  <div
    className="calendar-day is-prev_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    28
  </div>
  <div
    className="calendar-day is-prev_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="calendar-day is-prev_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="calendar-day is-prev_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    31
  </div>
  <div
    className="calendar-day is-current_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="calendar-day is-current_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
</div>
`

exports[`Week #render when prop \`date\` is different month renders <Week /> 1`] = `
<div
  className="calendar-week"
>
  <div
    className="calendar-day is-prev_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="calendar-day is-prev_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="calendar-day is-prev_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="calendar-day is-prev_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="calendar-day is-prev_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="calendar-day is-prev_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="calendar-day is-prev_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`

exports[`Week #render when prop \`dateDisabled\` renders <Week /> 1`] = `
<div
  className="cal-week"
>
  <div
    className="cal-day is-highlighted is-current_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="cal-day is-selected is-highlighted is-current_month is-start_selection is-end_selection is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="cal-day is-highlighted is-today is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="cal-day is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="cal-day is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="cal-day is-next_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="cal-day is-next_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`

exports[`Week #render when prop \`disabledIntervals\` is defined renders <Week /> 1`] = `
<div
  className="cal-week"
>
  <div
    className="cal-day is-highlighted is-current_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="cal-day is-selected is-highlighted is-current_month is-start_selection is-end_selection is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="cal-day is-highlighted is-today is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="cal-day is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="cal-day is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="cal-day is-next_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="cal-day is-next_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`

exports[`Week #render when prop \`highlightedStart\` and \`highlightedEnd\` are defined renders <Week /> 1`] = `
<div
  className="calendar-week"
>
  <div
    className="calendar-day is-highlighted is-current_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="calendar-day is-highlighted is-current_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="calendar-day is-highlighted is-today is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`

exports[`Week #render when prop \`maxDate\` is defined renders <Week /> 1`] = `
<div
  className="calendar-week"
>
  <div
    className="calendar-day is-highlighted is-current_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="calendar-day is-selected is-highlighted is-current_month is-start_selection is-end_selection is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="calendar-day is-highlighted is-today is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`

exports[`Week #render when prop \`minDate\` is defined renders <Week /> 1`] = `
<div
  className="calendar-week"
>
  <div
    className="calendar-day is-highlighted is-current_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="calendar-day is-selected is-highlighted is-current_month is-start_selection is-end_selection is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="calendar-day is-highlighted is-today is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`

exports[`Week #render when prop \`selectedMin\` and \`selectedMax\` are defined renders <Week /> 1`] = `
<div
  className="calendar-week"
>
  <div
    className="calendar-day is-highlighted is-current_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="calendar-day is-selected is-highlighted is-current_month is-start_selection is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="calendar-day is-selected is-highlighted is-today is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="calendar-day is-selected is-end_selection is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`

exports[`Week #render when prop \`selectedMin\` and \`selectedMax\` are defined renders <Week /> 2`] = `
<div
  className="calendar-week"
>
  <div
    className="calendar-day is-highlighted is-current_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="calendar-day is-selected is-highlighted is-current_month is-start_selection is-end_selection is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="calendar-day is-highlighted is-today is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-not_selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`

exports[
  `Week #render when prop \`selectedMin\` and \`selectedMax\` are defined when prop \`selectedMin\` and \`selectedMax\` are the same day renders <Week /> 1`
] = `
<div
  className="calendar-week"
>
  <div
    className="calendar-day is-highlighted is-current_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    29
  </div>
  <div
    className="calendar-day is-selected is-highlighted is-current_month is-start_selection is-end_selection is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    30
  </div>
  <div
    className="calendar-day is-highlighted is-today is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    1
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    2
  </div>
  <div
    className="calendar-day is-next_month is-working_day is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    3
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    4
  </div>
  <div
    className="calendar-day is-next_month is-weekend is-selectable"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    5
  </div>
</div>
`
